### DISCUSSION TITLE
How to get it running with ./run-example.sh simulation

### DISCUSSION THREAD

**Discussion by JohnAdam-123** [yamcs/yamcs#938] [General]
Hello, am new to Yamcs and I wanna know how to start this with "./run-example.sh simulation". I tried but the respond is "bash: ./run-example.sh: No such file or directory". I read yamcs get-started project but i couldn't get this running, any help please!?

**Comment by fqqb** [comment]
Hi, that script will not work in a clone of the quickstart project. The `run-example.sh` script belongs to the main Yamcs repository (where you made this issue). It is used during development of Yamcs. The configuration files (etc/mdb folders) may also be of interest outside of core development.

Ideally, we should move those examples somewhere else, so that you could use them in the same manner, as explained in the quickstart project (with `mvn yamcs:run`), but that's for another day.

If you do want to use `run-example.sh`, you should clone this repository, and follow the readme to compile Yamcs yourself.

**Comment by JohnAdam-123** [comment]
Okay, I did follow that readMe file. But I reached a point where I have to
run the simulation. Now after I run "./run-example.sh" it gives me option
to choose one is simulation, but the error am getting is " [ERROR] Failed
to execute goal on project simulation: Could not resolve dependencies for
project org.yamcs.examples:simulation:jar:5.10.2-SNAPSHOT: The following
artifacts could not be resolved: org.yamcs:simulator:jar:5.10.2-SNAPSHOT,
org.yamcs:yamcs-core:jar:5.10.2-SNAPSHOT,
org.yamcs:yamcs-tse:jar:5.10.2-SNAPSHOT,
org.yamcs:yamcs-web:jar:5.10.2-SNAPSHOT: Could not find artifact
org.yamcs:simulator:jar:5.10.2-SNAPSHOT -> [Help 1]"

Can you help?

Regards,
J. M. Adam

**Reply by JohnAdam-123** [reply]
> Okay, I did follow that readMe file. But I reached a point where I have to run the simulation. Now after I run "./run-example.sh" it gives me option to choose one is simulation, but the error am getting is " [ERROR] Failed to execute goal on project simulation: Could not resolve dependencies for project org.yamcs.examples:simulation:jar:5.10.2-SNAPSHOT: The following artifacts could not be resolved: org.yamcs:simulator:jar:5.10.2-SNAPSHOT, org.yamcs:yamcs-core:jar:5.10.2-SNAPSHOT, org.yamcs:yamcs-tse:jar:5.10.2-SNAPSHOT, org.yamcs:yamcs-web:jar:5.10.2-SNAPSHOT: Could not find artifact org.yamcs:simulator:jar:5.10.2-SNAPSHOT -> [Help 1]" Can you help? Regards, J. M. Adam
> […](#)
> On Mon, Sep 16, 2024 at 5:40 PM Fabian Diet ***@***.***> wrote: Hi, that script will not work in a clone of the quickstart project. The run-example.sh script belongs to the main Yamcs repository (where you made this issue). It is used during development of Yamcs. The configuration files (etc/mdb folders) may also be of interest outside of core development. Ideally, we should move those examples somewhere else, so that you could use them in the same manner, as explained in the quickstart project (with mvn yamcs:run), but that's for another day. If you do want to use run-example.sh, you should clone this repository, and follow the readme to compile Yamcs yourself. — Reply to this email directly, view it on GitHub <[#938 (comment)](https://github.com/yamcs/yamcs/discussions/938#discussioncomment-10660657)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/A6MHT44THCW6ENTOXC2HPCDZW3UU5AVCNFSM6AAAAABOJNHFN2VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANRWGA3DKNY> . You are receiving this because you authored the thread.Message ID: ***@***.***>

Am thinking i should add some dependencies maybe in pom file as configuring.

**Comment by fqqb** [comment]
Looks like you didn't compile Yamcs, which is required if you want to use the run-example.sh script.

I was referring to this readme: https://github.com/yamcs/yamcs/blob/master/README.md
Specifically, section: "Development Setup".

**Reply by JohnAdam-123** [reply]
I did read it and compiled the files. For example here are the code errors that  i get
"./run-example.sh
usage: ./run-example.sh EXAMPLE [options]

Where EXAMPLE is one of:
    cascading
    ccsds-frames
    cfdp
    cfdp-udp
    perftest1
    perftest2
    pus
    replication1
    replication2
    replication3
    simulation
    target
    templates
"
And here is what it gives as output...
" /run-example.sh templates
[INFO] Scanning for projects...
[WARNING] Failed to build parent project for org.yamcs.examples:examples:pom:5.10.2-SNAPSHOT
[WARNING] Failed to build parent project for org.yamcs.examples:templates:jar:5.10.2-SNAPSHOT
[INFO] 
[INFO] --------------------< org.yamcs.examples:templates >--------------------
[INFO] Building Yamcs :: Examples :: Templates 5.10.2-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] >>> yamcs-maven-plugin:1.3.2:run (default-cli) > process-classes @ templates >>>
[WARNING] The POM for org.yamcs:yamcs-core:jar:5.10.2-SNAPSHOT is missing, no dependency information available
[WARNING] The POM for org.yamcs:yamcs-web:jar:5.10.2-SNAPSHOT is missing, no dependency information available
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.067 s
[INFO] Finished at: 2024-09-16T19:06:54+03:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project templates: Could not resolve dependencies for project org.yamcs.examples:templates:jar:5.10.2-SNAPSHOT: The following artifacts could not be resolved: org.yamcs:yamcs-core:jar:5.10.2-SNAPSHOT, org.yamcs:yamcs-web:jar:5.10.2-SNAPSHOT: Could not find artifact org.yamcs:yamcs-core:jar:5.10.2-SNAPSHOT -> [Help 1]
"
So, does this mean I haven't compiled yet?

**Comment by fqqb** [comment]
> So, does this mean I haven't compiled yet?

Yes, i can reproduce your error under those conditions.

I think some basic must be going wrong. Please ensure the command is exactly `mvn clean install -DskipTests`, and that you run that from the top directory of your yamcs repository clone.

**Reply by JohnAdam-123** [reply]
Okay, I'll let you know if it worked out.
